src/main/java/Dude/Dude.java

```diff
@@ -0,0 +1,601 @@
+     *
+     * @return A string representation of the ToDo task.
+     */
+    @Override
+    public String toString() {
+        return "[T]" + (isDone ? "[X] " : "[ ] ") + description;
+    }
+}
+
+/**
+ * Represents a Deadline task with a description and a due date.
+ * Inherits from the Task class.
+ */
+class Deadline extends Task {
+    /**
+     * The due date of the Deadline task.
+     */
+    LocalDate by;
+
+    /**
+     * Constructs a new Deadline task with the specified description and due date.
+     *
+     * @param description The description of the Deadline task.
+     * @param by          The due date of the task in the format "yyyy-MM-dd".
+     */
+    public Deadline(String description, String by) {
+        super(description);
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+        this.by = LocalDate.parse(by, formatter);
+    }
+
+    /**
+     * Returns a string representation of the Deadline task,
+     * including its type, completion status, description, and due date.
+     *
+     * @return A string representation of the Deadline task.
+     */
+    @Override
+    public String toString() {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("MMM-dd-yyyy");
+        String formattedDate = by.format(formatter);
+        return "[D]" + (isDone ? "[X] " : "[ ] ") + description + " (by: " + formattedDate + ")";
+    }
+}
+
+
+/**
+ * Represents an Event task with a description, start date, and end date.
+ * Inherits from the Task class.
+ */
+class Event extends Task {
+    LocalDate start;
+    LocalDate end;
+    /**
+     * Constructs a new Event task with the specified description, start date, and end date.
+     *
+     * @param description The description of the Event task.
+     * @param start       The start date of the event in the format "yyyy-MM-dd".
+     * @param end         The end date of the event in the format "yyyy-MM-dd".
+     */
+    public Event(String description, String start, String end) {
+        super(description);
+        // Parse the strings to LocalDate
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+        this.start = LocalDate.parse(start, formatter);
+        this.end = LocalDate.parse(end, formatter);
+    }
+
+    /**
+     * Returns a string representation of the Event task,
+     * including its type, completion status, description, start date, and end date.
+     *
+     * @return A string representation of the Event task.
+     */
+    @Override
+    public String toString() {
+        // Define a formatter
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("MMM-dd-yyyy");
+        // Format the LocalDate to the desired format
+        String formattedStart = start.format(formatter);
+        String formattedEnd = end.format(formatter);
+        return "[E]" + (isDone ? "[X] " : "[ ] ") + description + " (from: " + formattedStart + " to: " + formattedEnd
+                + ")";
```

Comment:
I like the fact that you use verb for methods, and nouns for objects, although the word formatted can be a bit confusing.